WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -201,14 +199,22 @@ extension PointOfSaleAggregateModel {
         }
     }
 
-    func startCashPayment() {
-        isCashPaymentInProgress = true
+    // Prevents card payments from the moment the merchant decides we start collecting cash
+    // Once we get the callback from the card service, we switch to cash collection state
+    @MainActor
+    func startCashPayment() async {
+        try? await cardPresentPaymentService.cancelPayment()
         paymentState = .cash(.collectingCash)
     }
 
-    func cancelCashPayment() {
-        isCashPaymentInProgress = false
-        paymentState = .card(.idle)
+    @MainActor
+    func cancelCashPayment() async {
+        if case .connected = cardReaderConnectionStatus {
+            try? await cardPresentPaymentService.cancelPayment()
+            await collectPayment()
```

Comment:
It might be worth renaming this to `collectCardPayment` now, for clarity

Reply:
Agree! Done https://github.com/woocommerce/woocommerce-ios/pull/14795/commits/91b3b8d00b6bb2428d462e23759c999a5cb111a9